apps/condo/domains/news/components/RecipientCounter.tsx

```diff
@@ -0,0 +1,143 @@
+    } else {
+        if (newsItemScopes.length === 1 && isTargetedToEntireProperty(newsItemScopes[0])) {
+            return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInProperty' })
+        }
+
+        if (every(newsItemScopes, isTargetedToEntireProperty)) {
+            const displayCount = 3
+            const addressList = newsItemScopes.map(({ property }) => property.name).join(', ')
+            const andMoreCount = newsItemScopes.length <= displayCount ? null : newsItemScopes.length - displayCount
+            const andMore = !andMoreCount ? '' : intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInProperties.andMore' }, { count: andMoreCount })
+            return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInProperties' }, {
+                addressList,
+                andMore,
+            })
+        }
+
+        if (areTargetedToOneProperty(newsItemScopes) && every(newsItemScopes, isTargetedToUnitName)) {
+            const property: PropertyType = newsItemScopes[0].property
+            const targetedSections = detectTargetedSections(newsItemScopes, property)
+            if (targetedSections.length === 1) {
+                return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInPropertySection' }, {
+                    section: targetedSections[0].name,
+                    address: newsItemScopes[0].property.address,
+                })
+            } else if (targetedSections.length > 1) {
+                return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInPropertySections' }, {
+                    sections: map(targetedSections, 'name').join(', '),
+                    address: newsItemScopes[0].property.address,
+                })
+            } else {
+                return intl.formatMessage({ id: 'news.component.RecipientCounter.toResidentsInPropertyUnits' }, {
+                    unitNames: map(newsItemScopes, 'unitName').join(', '),
+                    address: newsItemScopes[0].property.address,
+                })
+            }
+        }
+    }
+}
+
+interface RecipientCounterProps {
+    newsItemScopes: NewsItemScope[]
+}
+
+export const RecipientCounter: React.FC<RecipientCounterProps> = ({ newsItemScopes }) => {
+    const intl = useIntl()
+    const MailingMessage = intl.formatMessage({ id: 'news.component.RecipientCounter.mailing' })
+    const PropertiesLabelMessage = intl.formatMessage({ id: 'news.component.RecipientCounter.label.properties' })
+    const WillReceiveLabelMessage = intl.formatMessage({ id: 'news.component.RecipientCounter.label.willReceive' })
+    const WillNotReceiveLabelMessage = intl.formatMessage({ id: 'news.component.RecipientCounter.label.willNotReceive' })
+
+    const { organization } = useOrganization()
+
+    const { objs: properties, loading: loadingProperties } = Property.useObjects({ where: { organization: { id: organization.id } } })
```

Comment:
This will load only the first 100 properties. Maybe it's better to use `Property.useCount()`?

Reply:
I need to use objects to calculate values of counters.

Reply:
> This will load only the first 100 properties. Maybe it's better to use `Property.useCount()`?

Where the restriction on first 100 items is implemented?

Reply:
My mistake, 1000, not 100
https://github.com/open-condo-software/condo/blob/0eae666329fa21964f436cb566ab05b97e6e7c78/packages/keystone/setup.utils.js#L78
Anyway, `useObjects` loads not all models. Maybe try `useAllObjects`?

Reply:
Done